homeassistant/components/script/__init__.py

```diff
@@ -129,6 +139,7 @@ def is_on(hass, entity_id):
 
     scripts = []
 
+    hass.data.setdefault("service_description_cache", {})
```

Comment:
We shouldn't mess with this cache ourselves. Instead, we should add a new method in `helpers/service.py` to set the schema for a service:

```python
@callback
def async_set_service_schema(domain, service, schema)
```

And we should use the same approach for python script integration.

Reply:
```python
@ha.callback
def async_set_service_schema(hass, domain, service, schema):
    """Register a description for a service."""
    hass.data.setdefault(SERVICE_DESCRIPTION_CACHE, {})

     description = {
        "description": schema.get("description", ""),
        "fields": schema.get("fields", {}),
    }

     hass.data[SERVICE_DESCRIPTION_CACHE]["{}.{}".format(domain, service)] = description
```

Is that what you had in mind?